tests/fixtures/mock_curl.php

```diff
@@ -27,15 +27,40 @@
 
 class mock_curl extends \curl {
 
-    public function __construct($responsecode, $response) {
+    /**
+     * Is this curl for testing retries?
+     *
+     * @var bool
+     */
+    private $retrytest;
```

Comment:
Are we able to rename this to `$shouldretry` so that it reads better when used in the conditions located in the post function? The description could be updated slightly too to better describe why the variable exists.